modules/firewall-rules/main.tf

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_compute_firewall" "rules" {
+  for_each                = { for r in var.rules : "${r.name}" => r }
+  name                    = each.value.name
+  description             = each.value.description
+  direction               = each.value.direction
+  network                 = var.network_name
+  project                 = var.project_id
+  source_ranges           = each.value.direction == "INGRESS" ? each.value.ranges : null
+  destination_ranges      = each.value.direction == "EGRESS" ? each.value.ranges : null
+  source_tags             = each.value.source_tags
+  source_service_accounts = each.value.source_service_accounts
+  target_tags             = each.value.target_tags
+  target_service_accounts = each.value.target_service_accounts
+  priority                = each.value.priority
```

Comment:
I saw `log_config` being passed through, but dont see it being set here?

Reply:
Good catch! This was an oversight on my part. I will fix this once I am done with the service identity features.

Reply:
Fixed.